src/users/users.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { CreateUserDto } from './dto/create-user.dto';
+import { UpdateUserDto } from './dto/update-user.dto';
+
+@Injectable()
+export class UsersService {
+  private users = []; 
+
+  create(createUserDto: CreateUserDto) {
+    const newUser = { id: Date.now(), ...createUserDto };
+    this.users.push(newUser);
```

Comment:
Next time please use only interfaces in services
in controller - DTO
in service - Interface